vortex-duckdb/src/convert/array/mod.rs

```diff
@@ -89,6 +92,50 @@ impl ToDuckDB for DictArray {
     }
 }
 
+pub fn selection_vector_from_array(prim: PrimitiveArray) -> SelectionVector {
+    match_each_integer_ptype!(prim.ptype(), |$P| {
+        selection_vector_from_slice(prim.as_slice::<$P>())
+    })
+}
+
+pub fn selection_vector_from_slice<P: NativePType + AsPrimitive<u32>>(
+    slice: &[P],
+) -> SelectionVector {
+    slice.iter().map(|v| (*v).as_()).collect()
+}
+
+const ALL_TRUE_SEL_MASK: [u64; 32] = [u64::MAX; 32];
+const ALL_FALSE_SEL_MASK: [u64; 32] = [0u64; 32];
+
+pub fn write_validity_from_mask(mask: Mask, flat_vector: &mut FlatVector) {
+    match mask {
+        Mask::AllTrue(_) => {
+            if let Some(slice) = flat_vector.validity_slice() {
+                // This is only needed if the vector is previously allocated.
+                bit_util::round_upto_multiple_of_64(flat_vector.capacity());
+                slice.copy_from_slice(
+                    &ALL_TRUE_SEL_MASK
+                        [0..bit_util::round_upto_multiple_of_64(flat_vector.capacity())],
```

Comment:
Don't we slice here from 0..to at least 64. But the selection masks only contain 32 elements?

Reply:
Ahhh I see

Reply:
Fixed, should be `ciel(cap/64)`